packages/appstore/src/constants/utils.ts

```diff
@@ -24,40 +25,8 @@ export const getHasDivider = (current_item_index: number, list_size: number, ava
 };
 
 // TODO: Moved to shared package! Delete it later, right now it uses for cashier wallet modals
-// TODO: Refactor using data transformation layer pattern when we will have API for wallets (e.g. wallet.icon)
-export const getWalletCurrencyIcon = (currency: string, is_dark_mode_on: boolean, is_modal = false) => {
-    switch (currency) {
-        case 'demo':
-            if (is_modal) return 'IcWalletDerivDemoLight';
-            return is_dark_mode_on ? 'IcWalletDerivDemoDark' : 'IcWalletDerivDemoLight';
-        case 'USD':
-            return 'IcWalletCurrencyUsd';
-        case 'EUR':
-            return 'IcWalletCurrencyEur';
-        case 'AUD':
-            return 'IcWalletCurrencyAud';
-        case 'GBP':
-            return 'IcWalletCurrencyGbp';
-        case 'BTC':
-            return is_dark_mode_on ? 'IcWalletBitcoinDark' : 'IcWalletBitcoinLight';
-        case 'ETH':
-            return is_dark_mode_on ? 'IcWalletEthereumDark' : 'IcWalletEthereumLight';
-        case 'USDT':
-        case 'eUSDT':
-        case 'tUSDT':
-        case 'UST':
-            if (is_modal) {
-                return is_dark_mode_on ? 'IcWalletModalTetherDark' : 'IcWalletModalTetherLight';
-            }
-            return is_dark_mode_on ? 'IcWalletTetherDark' : 'IcWalletTetherLight';
-        case 'LTC':
-            return is_dark_mode_on ? 'IcWalletLiteCoinDark' : 'IcWalletLiteCoinLight';
-        case 'USDC':
-            return is_dark_mode_on ? 'IcWalletUsdCoinDark' : 'IcWalletUsdCoinLight';
-        default:
-            return 'Unknown';
-    }
-};
+export const getWalletCurrencyIcon = (currency: string, is_dark_mode_on = false) =>
+    currency_to_icon_mapper[currency][`${is_dark_mode_on ? 'dark' : 'light'}`];
```

Comment:
let me explain what is happening here:
actually we have 3 places in our code base where we can get WalletCurrencyIcon: here (`appstore/srs/contsants/utils`), useWalletsList hook, and `@deriv/utils`.
In (`appstore/srs/contsants/utils`) and `@deriv/utils` there are exactly the same functions - `getWalletCurrencyIcon`
So at least from one place we can remove it. And I believe that it should be '@deriv/utils' since this function is specific to appstore packages.
So this is a good start for now.... The reason why i left two places: `useWalletsList` and `appstore/srs/contsants/utils` because in components where we use `WalletCurrencyName` we not always rely on existing wallets: (e.g) wallet-add-card.tsx component (we can't use icon from useWalletList because there are only wallets that will be created).
but :peach:
as i mentioned, this i is a good start point. so we can later discuss how and where we can store WalletCurrencyIcon.

Reply:
@heorhi-deriv - Are you able to add a "Tech Debt" story to the backlog to finish this off? That's the best practice otherwise we'll just forget about it. 

Good start!

Reply:
@heorhi-deriv I'm not sure about currency_to_icon_mapper and getWalletCurrencyIcon to be only in appstore package. I mean we use the same in useWalletsList and useWalletAccountsList hooks (inside of them), so maybe would be better to create these functions in @deriv/utils, because we have access there from appstore and from hooks packages. Just a question :) You can ignore it

Reply:
sorted